### PR TITLE
fix:restrict access to page for unauthorize user

### PR DESCRIPTION
**Pull request by sunilsabatp**
Fixes # some features are only available to tpo (i.e `profile/treemapper/import`) , restricted that pages to normal user

### CODE REVIEW

**Comment by mohitb35** [comment]
@sunilsabatp The PR looks good overall. 

Could you check if there are any other pages where we can do this - e.g. /profile/payouts and it's subpages don't show an access denied component but just redirect to /profile?

**Comment by sagararyal** [comment]
#### 📝 Changed routes:


| [**…youts/add-bank-details**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/add-bank-details) | [**/profile/payouts**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts) |
|:-:|:-:|
|<a href="https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/add-bank-details">Screenshot of /profile/payouts/add-bank-details</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/FQmrDdrGuCpU2mau7lY_uc_SqOZpfQEpxhpfBn2hvRShaAdwT8GOB4AyHgMupbJ6NB3OdKAQyC178Dn07dikKQ.png">(view full size)</a> |<a href="https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts">Screenshot of /profile/payouts</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/iHsEhrvcq19WjD_qdkqlhnIyiDQemX85GsDk7FK0iXOSWJ3NMWqxxx37_VE1zfQ_nHQdoDPwGfyKzbZkBl10SQ.png">(view full size)</a> |



| [**/profile/payouts/schedule**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/schedule) | [**/profile/projects**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects) |
|:-:|:-:|
|<a href="https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/schedule">Screenshot of /profile/payouts/schedule</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/L9oQQwWljMxwkdj16Etx8rFWxcv5DzWKiLKgiX7eAJH-2SYbx8xK1WQUX6tniqWRXluUtXPQvBPJIw5R6PntFQ.png">(view full size)</a> |<a href="https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects">Screenshot of /profile/projects</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/JLvoU09cJAA0HpaTVoqnNP_J_lETVbUX2n6cZLMCYW3py4ezDGIvTeaohQWut8cndMi0wmeph85X43witigGJw.png">(view full size)</a> |



| [**…e/projects/new-project**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects/new-project) | [**…file/treemapper/import**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/treemapper/import) |
|:-:|:-:|
|<a href="https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects/new-project">Screenshot of /profile/projects/new-project</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/O-btVHtIVL-KLsACYWdY9JKDEY0ChX23LSqSh-X5QPlQNnJ56hLU1d3-oxYC3Zo8aWL7IgLPw3D0gCU2Vt53QA.png">(view full size)</a> |<a href="https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/treemapper/import">Screenshot of /profile/treemapper/import</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/jHGPz9TF6_cbT5HNJqYpyusZ_YlH3raoASxOKHe9q9GOGHvVaNxACLKvE_GnGiuQP3cR5pL9fh9lIooTONuQ1A.png">(view full size)</a> |



And 1 other route:
- [**/profile/treemapper/my-species**](https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/treemapper/my-species)



Commit b6ee743fa955cac27e0351419be3ae29e3fed59f (https://planet-webapp-q3ufbefmq-planetapp.vercel.app).

**Review by mohitb35 — APPROVED** [review]
LGTM

**Comment by mariahosfeld** [comment]
Functionality works for me as well.

**Review by norbertschuler — APPROVED** [review]
LGFM. I tested the pages for a TPO account and afterwards tried to access them with a non TPO account:

- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/schedule
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/add-bank-details
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/payouts/edit-bank-details/roacc_s1g4aD2k8NmU
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects/new-project
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects/new-project?purpose=conservation
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects/proj_GOSwPC8lawTDjM3W8PnNP3b3?type=basic-details
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/projects/site_nIkV8dPS8WpXpM2
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/treemapper/import
- https://planet-webapp-q3ufbefmq-planetapp.vercel.app/profile/treemapper/my-species

File: pages/profile/treemapper/my-species.tsx [hunk 1]
```diff
@@ -1,20 +1,23 @@
...
 export default function MySpeciesPage({}: Props): ReactElement {
   const { t } = useTranslation('me');
+  const { user } = useContext(UserPropsContext);
   return (
     <UserLayout>
       <Head>
         <title>{t('mySpecies')}</title>
       </Head>
-      <MySpecies />
+      {user.type === 'tpo' ? <MySpecies /> : <AccessDeniedLoader />}
```
Comment (norbertschuler) [inline pages/profile/treemapper/my-species.tsx:20]: `AccessDeniedLoader` is not translated, but I don't mind.

Comment (norbertschuler) [reply pages/profile/treemapper/my-species.tsx:20]: Requested translation at https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1672#discussion_r1110984590

**Pull request merged**